From 18c4d8875cbb6f8c691c5a96f50517ef6fe87960 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Arthur=20Exalta=C3=A7=C3=A3o?=
 <arthur.exaltacao@chainguard.dev>
Date: Tue, 19 Aug 2025 10:56:14 -0300
Subject: [PATCH] freeipa: ensure detected krb5 version does not have -final
 suffix
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This is just not expected by freeIPA and will make it so anything that
uses krb5 versioning breaks

Signed-off-by: Arthur Exaltação <arthur.exaltacao@chainguard.dev>
---
 VERSION.m4              | 2 +-
 ipapython/version.py.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/VERSION.m4 b/VERSION.m4
index 20aa1a241..f013944e3 100644
--- a/VERSION.m4
+++ b/VERSION.m4
@@ -140,7 +140,7 @@ NEWLINE)) dnl IPA_VERSION end
 ########################################################
 # Version of MIT krb5 used to build IPA
 ########################################################
-define(IPA_KRB5_BUILD_VERSION, translit(esyscmd(krb5-config --version | awk '{ print $NF }'), NEWLINE))
+define(IPA_KRB5_BUILD_VERSION, translit(esyscmd(krb5-config --version | sed 's/-final//' | awk '{ print $NF }'), NEWLINE))
 
 dnl DEBUG: uncomment following lines and run command m4 VERSION.m4
 dnl `IPA_VERSION: ''IPA_VERSION'
diff --git a/ipapython/version.py.in b/ipapython/version.py.in
index eee8900be..5cef9854d 100644
--- a/ipapython/version.py.in
+++ b/ipapython/version.py.in
@@ -54,4 +54,4 @@ DEFAULT_PLUGINS = frozenset(l.strip() for l in """
 @DEFAULT_PLUGINS@
 """.strip().splitlines())
 
-KRB5_BUILD_VERSION = parse_version("@KRB5_BUILD_VERSION@")
+KRB5_BUILD_VERSION = parse_version("@KRB5_BUILD_VERSION@".replace("-final", ""))
-- 
2.50.1


